### ISSUE TITLE
VORG in CFF2 variable font or TT

### ISSUE THREAD

**Issue by blueshade7** [harfbuzz/harfbuzz#1694]
`hb_ot_get_glyph_v_origin()` in `hb-ot-font.cc` reads the vertical origin from the VORG table if it is present, then returns the value.

Two issues:
If it is a CFF2 variable font, a delta value from the VVAR table must be applied to the default value.
If it is a TrueType font, the VORG table must be ignored. OpenType spec explicitly says so.

**Comment by behdad** [comment]
I don't care about the latter part.  If VORG is there, I prefer to respect it.

**Comment by behdad** [comment]
But sure, should be trivial to move the 'glyf' code before the 'VORG' code.

**Comment by behdad** [comment]
Thanks. I'll see about fixing this.

**Comment by behdad** [comment]
I fixed this. The ttf and otf get close-enough numbers now, but not identical. @punchcutter is that expected?

**Comment by behdad** [comment]
Reopening to confirm, also to add tests.

**Comment by punchcutter** [comment]
> I fixed this. The ttf and otf get close-enough numbers now, but not identical. @punchcutter is that expected?

No, they should be the same. With these changes the VORG variation seems to be good, but now I see that the variation in the TTF was also off in a different way. I hadn't noticed at first because I was focused on the VORG. In this example I don't see any variation happening in tsb. It's only using the default value 88, but doesn't get any delta. I checked the gvar to be sure and the glyf extents delta is there as 210 which is the difference between extents.y_bearing = 678 at wdth=50 and 888 at wdth=150. That should give a tsb of 106 at wdth=150, but that's not happening.

**Comment by behdad** [comment]
I think I see the bug.

**Comment by behdad** [comment]
> That should give a tsb of 106 at wdth=150, but that's not happening.

But there's also a deltaset moving the top phantom point by -228 points, pushing the tsb to something very different.

**Comment by behdad** [comment]
@punchcutter
I created https://github.com/harfbuzz/harfbuzz/pull/3702 with some more fixed. Please check it out.
As I said, the TrueType font now gets very off values because the phantom point is moved far away.

**Comment by behdad** [comment]
> But there's also a deltaset moving the top phantom point by -228 points, pushing the tsb to something very different.

There seems to be a bug at least, that -228 seems to be applied twice...

**Comment by behdad** [comment]
> There seems to be a bug at least, that -228 seems to be applied twice...

I'm not sure. Compared to the otf position I'm seeing 2*-228 difference. But I can't see any problem in the code. It seems to follow what's in the tables.

**Comment by behdad** [comment]
If I modify the ttf file and change the delta for the phantom point to be +228 instead of -228, then I get the same result as the otf. The positive sounds correct to me. @punchcutter wdyt? How did you create the gvar table?

**Comment by punchcutter** [comment]
https://github.com/harfbuzz/harfbuzz/pull/3702 looks good to me now. The master locations all match, but there are some off by one in the interpolation. That doesn't suprise me too much because one is using VORG values and the other is using vmtx and glyf extents. I'm not sure it would make sense for them to be exactly the same for every value anyway. 

I made one mistake in the original TTF I attached. It shares a lot of tables with the OTF and I accidentally gave it the VVAR from the OTF, but either way I get the same results with your fixes. Updated TTF with fontTools built VVAR: [VVARTest.ttf.zip](https://github.com/harfbuzz/harfbuzz/files/9034121/VVARTest.ttf.zip) The gvar is from fontTools latest code on main. The only other difference is I added vmtx parsing to feaLib locally (getting ready to make a PR).
